daemon/cmd/config.go

```diff
@@ -80,6 +80,11 @@ func (h *patchConfig) Handle(params PatchConfigParams) middleware.Responder {
 	}
 
 	changes += option.Config.Opts.ApplyValidated(om, changedOption, d)
+	err = d.endpointManager.OverrideEndpointOpts(om)
+	if err != nil {
+		msg := fmt.Errorf("Override endpoint configuration failed %s", err)
+		return api.Error(PatchConfigFailureCode, msg)
+	}
```

Comment:
Strictly speaking, this has an unintended impact if some portion of the API request fails: If the above is successful, but then line 95 sees an error.. then the endpoint options will have been configured, but the changes will not have been applied yet. Then the API will return an error, but the changes were already made! Then next time someone tries to regenerate the endpoints for any reason, the changes may take effect.

There's two broad ways to solve this:
1. Undo these changes in the error condition below. One way we have done this in the past is to have the function like `OverrideEndpointOpts()` return a "revert function" (see `revertFunc') which can be run in the error condition below.
1. Move the configuration logic into the per-endpoint eventqueues for modifying endpoint state.

Reply:
You are right, agent option changes should be reverted on error as well.